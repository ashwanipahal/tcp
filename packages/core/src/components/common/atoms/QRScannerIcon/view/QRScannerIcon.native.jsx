import React from 'react';
import CustomIcon from '../../Icon';
import { ICON_NAME, ICON_FONT_CLASS } from '../../Icon/Icon.constants';

const onQRIconFocus = (navigation, closeModal) => {
  if (navigation) {
    navigation.navigate('QRScanner');
  }
  if (closeModal) {
    closeModal();
  }
};

const QRScannerIcon = ({ navigation, closeModal, style, size }) => {
  return (
    <CustomIcon
      onPress={() => onQRIconFocus(navigation, closeModal)}
      iconFontName={ICON_FONT_CLASS.Icomoon}
      name={ICON_NAME.qrcode}
      size={size}
      color="gray.600"
      accessibilityLabel="barcode"
      style={style}
      isButton
    />
  );
};

export default QRScannerIcon;
