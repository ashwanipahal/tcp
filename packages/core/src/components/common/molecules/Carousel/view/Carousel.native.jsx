import React from 'react';
import { TouchableOpacity, View } from 'react-native';
import Carousel from 'react-native-snap-carousel';
import { Image } from '../../../atoms';
import config from '../config';
import theme from '../../../../../../styles/themes/TCP';

const playIcon = require('../../../../../../assets/play.png');
const pauseIcon = require('../../../../../../assets/pause.png');

// @flow
type Props = {
  carouselConfig: Object,
  data: Array<Object>,
  renderItem: Function,
  onSnapToItem: Function,
  width: Number,
  height: Number,
  slideStyle: Object,
};

type State = {
  autoplay: Boolean,
};

/**
 * Default settings for Carousel.
 */
const defaults = { ...config.CAROUSEL_MOBILE_DEFAULTS };

/**
 * Style for play pause icons.
 */
const { colors, zindex } = theme;
const { playIconHeight, playIconWidth } = config.CAROUSEL_MOBILE_CONFIG;
const touchableStyle = {
  backgroundColor: colors.WHITE,
  borderRadius: 15,
  position: 'absolute',
  bottom: 12,
  left: '50%',
  height: playIconHeight,
  width: playIconWidth,
  transform: [
    {
      translateX: -15,
    },
  ],
  zIndex: zindex.zOverlay,
};

/**
 * @function Carousel component that creates carousel using
 * third party library 'react-native-snap-carousel'
 */
class SnapCarousel extends React.PureComponent<Props, State> {
  constructor(props) {
    super(props);
    this.state = {
      autoplay: true,
    };
    this.getPlayButton = this.getPlayButton.bind(this);
    this.play = this.play.bind(this);
    this.pause = this.pause.bind(this);
  }

  /**
   * @function getPlayComponent function gets DOM reference of slider component.
   * @param {[Object]} element [Event object of click].
   * @return {node} function returns slider element.
   */
  getPlayButton() {
    const { autoplay } = this.state;
    return autoplay ? (
      <TouchableOpacity accessibilityRole="button" onPress={this.pause} style={touchableStyle}>
        <Image source={pauseIcon} height={playIconHeight} width={playIconWidth} />
      </TouchableOpacity>
    ) : (
      <TouchableOpacity accessibilityRole="button" onPress={this.play} style={touchableStyle}>
        <Image source={playIcon} height={playIconHeight} width={playIconWidth} />
      </TouchableOpacity>
    );
  }

  /**
   * @function play function enable autoplay for carousel
   * also update component state.
   */
  play() {
    this.carousel.startAutoplay();
    this.togglePlay();
  }

  /**
   * @function pause function stops/pause autoplay for carousel
   * also update component state.
   */
  pause() {
    this.carousel.stopAutoplay();
    this.togglePlay();
  }

  /**
   * @function togglePlay function updates component state
   * after each tap of play pause button.
   */
  togglePlay() {
    const { autoplay } = this.state;
    this.setState({ autoplay: !autoplay });
  }

  render() {
    const {
      carouselConfig: { autoplay: defaultAutoplay },
      data,
      height,
      width,
      onSnapToItem,
      renderItem,
      slideStyle,
    } = this.props;

    return (
      <View>
        <Carousel
          ref={c => {
            this.carousel = c;
          }}
          onSnapToItem={onSnapToItem}
          data={data}
          renderItem={renderItem}
          sliderWidth={width}
          itemWidth={width}
          sliderHeight={height}
          itemHeight={height}
          slideStyle={slideStyle}
          {...defaults}
        />
        {defaultAutoplay && this.getPlayButton()}
      </View>
    );
  }
}

export default SnapCarousel;
