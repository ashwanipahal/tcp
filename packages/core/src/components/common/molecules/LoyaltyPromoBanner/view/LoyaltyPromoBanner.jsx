import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import ClickTracker from '@tcp/web/src/components/common/atoms/ClickTracker';
import { Anchor, RichText, Row, Col } from '../../../atoms';
import { readCookie, setCookie } from '../../../../../utils/cookie.util';
import { loyalityAnalyticsValue } from '../../../../../constants/analytics';
import withStyles from '../../../hoc/withStyles';

import style from '../LoyaltyPromoBanner.style';

/**
 * Check if required UUID cookie available else return default.
 */
const getUUID = uuidCookieString => {
  const UUID = readCookie(uuidCookieString);
  return UUID ? UUID.split(',')[0] : '-1002';
};

/* istanbul ignore next */
const LoyaltyPromoBanner = props => {
  const {
    className,
    cookieID,
    richTextList: [{ richText, link }],
    dataLocator,
  } = props;
  const cookieName = `${cookieID}_${getUUID('WC_USERACTIVITY_')}`;
  const [bannerClosed, setBannerClosed] = useState(true);

  useEffect(() => {
    if (!readCookie(cookieName)) {
      setBannerClosed(false);
    }
  }, [bannerClosed]);

  const closeButtonHandler = () => {
    const currentDate = new Date();
    setCookie({
      key: cookieName,
      value: currentDate.toGMTString(),
      daysAlive: 10,
    });
    setBannerClosed(true);
  };

  return bannerClosed ? null : (
    <div className={`${className} content-wrapper`}>
      <Row>
        <Col
          colSize={{
            small: 6,
            medium: 8,
            large: 12,
          }}
        >
          {link ? (
            <ClickTracker
              as={Anchor}
              to={link.url}
              asPath={link.url}
              target={link.target}
              title={link.title}
              dataLocator={dataLocator || `loyalty-promo-banner`}
              clickData={{
                customEvents: ['event80', 'event81'],
                internalCampaignId: loyalityAnalyticsValue,
              }}
            >
              <RichText richTextHtml={richText.text} />
            </ClickTracker>
          ) : (
            <RichText richTextHtml={richText.text} />
          )}
          <button
            aria-label="close"
            className="loyalty-promo-close-btn"
            onClick={closeButtonHandler}
          >
            <svg
              className="loyalty-promo-close-btn-icon"
              viewBox="0 0 25 25"
              aria-hidden="true"
              role="img"
            >
              <path
                fill="#a0a0a0"
                fillRule="nonzero"
                d="M14.107 12.5l10.56-10.56A1.136 1.136 0 1 0 23.06.333L12.5 10.893 1.94.333A1.136 1.136 0 1 0 .333 1.94l10.56 10.56L.333 23.06a1.136 1.136 0 1 0 1.607 1.607l10.56-10.56 10.56 10.56c.222.222.513.333.804.333a1.136 1.136 0 0 0 .803-1.94L14.107 12.5z"
              />
            </svg>
          </button>
        </Col>
      </Row>
    </div>
  );
};

LoyaltyPromoBanner.propTypes = {
  className: PropTypes.string.isRequired,
  cookieID: PropTypes.string.isRequired,
  richTextList: PropTypes.arrayOf(PropTypes.object),
  dataLocator: PropTypes.string,
};

LoyaltyPromoBanner.defaultProps = {
  richTextList: [
    {
      richText: { text: '' },
      link: {},
    },
  ],
  dataLocator: '',
};

export default withStyles(LoyaltyPromoBanner, style);
export { LoyaltyPromoBanner as LoyaltyPromoBannerVanilla };
