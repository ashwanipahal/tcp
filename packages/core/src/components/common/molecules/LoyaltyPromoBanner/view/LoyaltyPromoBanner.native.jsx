import React from 'react';
import { View } from 'react-native';
import AsyncStorage from '@react-native-community/async-storage';
import PropTypes from 'prop-types';
import { RichText, Anchor } from '../../../atoms';
import { Wrapper, StyledImage, Touchable } from '../LoyaltyPromoBanner.style.native';

const crossImage = require('../../../../../../src/assets/close.png');

const cookieName = '@mprAboveHead_-1002:key';
const date = new Date();
const currentDateValue = date.getTime();
const daysAlive = 10;

class LoyaltyPromoBanner extends React.PureComponent {
  /**
   * To manage the state of icons on the
   * basis of visible & hide .
   */
  constructor(props) {
    super(props);
    this.state = {
      bannerClosed: false,
    };
  }

  componentDidMount() {
    this.isDisplay();
  }

  /**
   * To save the date
   */
  setDate = () => {
    const expiteDateValue = date.setTime(date.getTime() + daysAlive * 24 * 60 * 60 * 1000);
    try {
      AsyncStorage.setItem(cookieName, expiteDateValue.toString());
    } catch (error) {
      // eslint-disable-next-line rule
      console.info('error', error);
    }
  };

  /**
   * To fetch the date
   */
  retrieveDate = async () => {
    try {
      return await AsyncStorage.getItem(cookieName);
    } catch (error) {
      // Error retrieving data
      return null;
    }
  };

  /**
   * To check the visiblity of the banner
   */
  isDisplay = () => {
    this.retrieveDate().then(data => {
      if (data !== null) {
        if (currentDateValue <= parseInt(data, 10)) {
          this.setState({
            bannerClosed: true,
          });
        } else {
          this.setState({
            bannerClosed: false,
          });
        }
      }
    });
  };

  /**
   * This function validate the iconView.
   */
  validateView = () => {
    this.setDate();
    this.setState({ bannerClosed: true });
  };

  /**
   * To render the view
   */
  render() {
    const {
      richTextList: [{ richText, link }],
      navigation,
    } = this.props;

    const { bannerClosed } = this.state;

    return (
      <View>
        {bannerClosed ? (
          <View />
        ) : (
          <Anchor navigation={navigation} url={link.url}>
            <Wrapper>
              <RichText
                source={{
                  html: richText.text,
                }}
              />
              <Touchable
                accessibilityRole="button"
                accessibilityLabels="close"
                onPress={this.validateView}
              >
                <StyledImage source={crossImage} />
              </Touchable>
            </Wrapper>
          </Anchor>
        )}
      </View>
    );
  }
}

LoyaltyPromoBanner.propTypes = {
  richTextList: PropTypes.arrayOf(PropTypes.object),
  navigation: PropTypes.shape({}).isRequired,
};

LoyaltyPromoBanner.defaultProps = {
  richTextList: [
    {
      richText: { text: '' },
      link: {},
    },
  ],
};

export default LoyaltyPromoBanner;
export { LoyaltyPromoBanner as LoyaltyPromoBannerVanilla };
