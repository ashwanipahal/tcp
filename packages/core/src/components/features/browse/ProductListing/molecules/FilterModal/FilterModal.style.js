import { css } from 'styled-components';

export default css`
  &.modal {
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background: rgba(0, 0, 0, 0.6);
    z-index: ${props => props.theme.zindex.zPLPFilterDropDown};
  }

  &.display-block {
    display: block;
  }

  &.display-none {
    display: none;
  }

  .modal-spacing {
    margin-bottom: ${props => props.theme.spacing.APP_LAYOUT_SPACING.SM};
    margin-top: 22px;

    button {
      max-width: 164px;
      height: 42px;
    }

    .close-button {
      text-align: right;
    }
  }

  .modal-main {
    position: fixed;
    background: white;
    width: 100%;
    height: auto;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    padding: 27px;

    @media ${props => props.theme.mediaQuery.medium} {
      width: 80%;
      max-width: 600px;
    }

    .filter-row {
      margin-bottom: 7px;
      margin-top: 14px;
    }

    .accordion {
      &.active:after,
      &.inactive:after {
        right: 42px;
      }
    }
  }

  .open-filter-button {
    background: url('/static/images/carrot-small-up.png') no-repeat;
    background-position: 95% 18px;
    background-color: ${props => props.theme.colors.ACCORDION.INACTIVE_HEADER};
    font-size: ${props => props.theme.fonts.fontSize.body.large.secondary}px;

    &.open-filter-button-expanded {
      background: url('/static/images/carrot-small-down.png') no-repeat;
      background-position: 95% 18px;

      &:hover:not([disabled]) {
        background: url('/static/images/carrot-small-down.png') no-repeat;
        background-position: 95% 18px;
        background-color: ${props => props.theme.colors.PRIMARY.LIGHTGRAY};
      }
    }

    &:hover:not([disabled]) {
      background: url(/static/images/carrot-small-up.png) no-repeat;
      background-position: 95% 18px;
      background-color: ${props => props.theme.colors.PRIMARY.LIGHTGRAY};
    }
  }
`;
