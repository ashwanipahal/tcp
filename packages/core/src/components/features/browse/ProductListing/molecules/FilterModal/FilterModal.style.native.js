import styled, { css } from 'styled-components/native';

const Container = styled.View`
  flex: 1;
`;

const ModalContainer = styled.View`
  flex: 1;
  background: rgba(26, 26, 26, 0.4);
  width: 100%;
  padding-top: 100;
`;

// margin-top: ${props => props.theme.spacing.ELEM_SPACING.XS};
const styles = css``;

export { styles, Container, ModalContainer };
