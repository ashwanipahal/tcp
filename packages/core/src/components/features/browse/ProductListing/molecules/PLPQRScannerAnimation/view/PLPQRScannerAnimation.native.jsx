import React from 'react';
import { PropTypes } from 'prop-types';
import { Dimensions } from 'react-native';
import ModalNative from '@tcp/core/src/components/common/molecules/Modal';
import FastImageNative from '../../../../../../../../../mobileapp/src/components/common/molecules/FastImage';

const closeIcon = require('@tcp/core/src/assets/close.png');

/**
 *
 * @param {*} navigation
 * @param {*} closeModal
 *
 * closeIconAction calls when user click on close icon of modal
 * navigation is used to take user to back on scanner page while
 * closeModal close the modal.
 *
 */
const closeIconAction = (navigation, closeModal) => {
  if (navigation) {
    navigation.navigate('QRScanner');
  }
  if (closeModal) {
    closeModal();
  }
};

const dimensions = Dimensions.get('screen');

/**
 * Fast image not render style component. Hence passing as style object
 */
const fastImageStyle = {
  width: dimensions.width - 28,
  height: dimensions.height - 290,
  marginTop: 100,
  marginRight: 'auto',
  marginBottom: 100,
  marginLeft: 'auto',
};

/**
 *
 * @param {*} url
 * @param {*} navigation
 * @param {*} resetCustomLoader
 * @param {*} isOpen
 *
 * This function render Given image URL in modal.
 * When user click on close icon it will take user to QRScanner page.
 */
const PLPQRScannerAnimation = ({ url, navigation, resetCustomLoader, isOpen }) => {
  return (
    <ModalNative
      rightAlignCrossIcon
      isOpen={isOpen}
      fullWidth
      horizontalBar={false}
      onRequestClose={() => closeIconAction(navigation, resetCustomLoader)}
      stickyCloseIcon
    >
      <FastImageNative url={url} style={fastImageStyle} />
    </ModalNative>
  );
};

PLPQRScannerAnimation.propTypes = {
  url: PropTypes.string,
  navigation: PropTypes.shape({}),
  resetCustomLoader: PropTypes.func,
  isOpen: PropTypes.bool,
};

PLPQRScannerAnimation.defaultProps = {
  url: '',
  navigation: {},
  resetCustomLoader: () => {},
  isOpen: false,
};

export default PLPQRScannerAnimation;
