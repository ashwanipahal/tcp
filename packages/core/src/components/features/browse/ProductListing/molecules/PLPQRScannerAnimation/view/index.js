export { default } from './PLPQRScannerAnimation.native';
