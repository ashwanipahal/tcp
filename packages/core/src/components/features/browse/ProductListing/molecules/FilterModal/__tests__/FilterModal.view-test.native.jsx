import React from 'react';
import { shallow } from 'enzyme';
import { FilterModalVanilla } from '../views/FilterModal.view';

describe('FilterModal is shown', () => {
  let wrapper;
  beforeEach(() => {
    const props = {
      filters: {},
      theme: {
        colorPalette: {
          text: {
            lightgray: '#eeeeee',
          },
        },
      },
      labelsFilter: {},
      onSubmit: () => {},
    };

    wrapper = shallow(<FilterModalVanilla {...props} />);
  });

  it('should render FilterModal when show is false', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should show filter modal', () => {
    const onSubmit = jest.fn();
    wrapper.setProps({
      onSubmit,
      navigation: {
        getParam: () => '',
      },
      setSelectedFilter: jest.fn(),
    });

    wrapper.instance().setState({ showSortModal: false });
    const modal = wrapper.find('[name="filters"]');
    modal.props().onSubmit({});
    expect(onSubmit).toHaveBeenCalled();
  });

  it('should close modal on close', () => {
    wrapper.instance().onCloseModal();
    expect(wrapper.instance().state.showModal).toBeFalsy();
  });

  it('should close modal on press out', () => {
    wrapper.instance().onPressOut();
    expect(wrapper.instance().state.showModal).toBeFalsy();
  });

  it('should open modal on press filter', () => {
    wrapper.instance().onPressFilter();
    expect(wrapper.instance().state.showModal).toBeTruthy();
  });

  it('should open modal on press sort', () => {
    wrapper.instance().onPressSort();
    expect(wrapper.instance().state.showModal).toBeTruthy();
  });
});
