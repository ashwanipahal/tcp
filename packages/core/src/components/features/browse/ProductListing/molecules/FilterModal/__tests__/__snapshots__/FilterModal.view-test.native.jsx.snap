// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterModal is shown should render FilterModal when show is false 1`] = `
<Styled(View)>
  <Styled(FilterButtons)
    labelsFilter={Object {}}
    onPressFilter={[Function]}
    onPressSort={[Function]}
    props={Object {}}
    selected={false}
  />
  <Component
    animationType="slide"
    hardwareAccelerated={false}
    transparent={true}
    visible={false}
  >
    <Styled(SafeAreaView)>
      <Styled(TouchableWithoutFeedback)
        accessibilityRole="button"
        activeOpacity={1}
        onPressOut={[Function]}
      >
        <Styled(View) />
      </Styled(TouchableWithoutFeedback)>
      <Styled(View)>
        <Styled(View)>
          <Styled(Text) />
          <Styled(TouchableOpacity)
            accessibilityLabel="Close button"
            accessibilityRole="button"
            isButton={true}
            onPress={[Function]}
          >
            <Styled(CustomIcon)
              accessibilityLabel=""
              color="gray.900"
              dataLocator=""
              iconFontName="FontAwesome"
              isButton={false}
              isDisabled={false}
              margins={null}
              name="close"
              onPress={[Function]}
              size="fs20"
            />
          </Styled(TouchableOpacity)>
        </Styled(View)>
        <Styled(WithTheme(Filters))
          filters={Object {}}
          labelsFilter={Object {}}
          name="filters"
          onSubmit={[Function]}
        />
      </Styled(View)>
    </Styled(SafeAreaView)>
  </Component>
</Styled(View)>
`;
